modules/thermo-cycler/production/eepromWriter/eepromWriter.ino

```diff
@@ -0,0 +1,215 @@
+#include <Wire.h>
+
+#define EEPROM_ADDR       0x52
+
+#define MAX_IN_NUM_LEN    17  // max serial = 17 (TCV01yyyymmddAnnn) | max model = 3 //vnn
+#define SERIAL_LOC        10
+#define BUFFER_BYTES      10
+#define MODEL_LOC         SERIAL_LOC + MAX_IN_NUM_LEN + BUFFER_BYTES
+#define VER_NUM_INDEX     3
+#define VER_NUM_LEN       2
+#define WR_TIME           5 //ms
+
+#define WP_PIN                  26
+#define PIN_FAN_SINK_CTRL       A4   // uses PWM frequency generator
+#define PIN_FAN_SINK_ENABLE     2    // Heat sink fan
+#define PIN_FRONT_BUTTON_LED    24
+#define PIN_PELTIER_ENABLE      7
+#define PIN_MOTOR_CURRENT_VREF  A0
+
+bool options_onscreen;
+
+bool eeprom_write_char(byte word_address, byte value)
+{
+  Wire.beginTransmission(EEPROM_ADDR);
+  Wire.write(word_address);
+  Wire.write(value);
+  byte error = Wire.endTransmission();
+  delay(WR_TIME);
+  if (error)
+  {
+    return false;
+  }
+  return true;
+}
+
+char eeprom_read_char(byte word_address)
+{
+  Wire.beginTransmission(EEPROM_ADDR);
+  Wire.write(word_address);
+  byte error = Wire.endTransmission();
+  if (!error)
+  {
+    Wire.requestFrom(EEPROM_ADDR, 1); // request 1 byte from eeprom
+    if (Wire.available())    // slave may send less than requested
+    {
+      char c = Wire.read(); // receive a byte as character
+      return c;         // print the character
+    }
+  }
+  return '~';
+}
+
+bool write_to_eeprom(String serial_num)
+{
+  char the_number[MAX_IN_NUM_LEN];
+  uint8_t number_len;
+  serial_num.toCharArray(the_number, serial_num.length()+1);
```

Comment:
```suggestion
  serial_num.toCharArray(the_number, MAX_IN_NUM_LEN);
```

Per [the arduino docs](https://www.arduino.cc/reference/en/language/variables/data-types/string/functions/tochararray/) that should be the size of the dest buffer.